### PR TITLE
Merged + unit tests

### LINKED ISSUE #138: Combined extends JsonResource
Let's add this and use it where we need to combine the user's input JsonObject with the API's output JsonOBject (e.g. when creating a Container, or a Swarm etc)

### PR DESCRIPTION
**Pull request by amihaiemil**
Fixes #138

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#139](https://github.com/amihaiemil/docker-java-api/issues/139) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#139](https://github.com/amihaiemil/docker-java-api/issues/139) is assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/1641147249); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/amihaiemil/docker/JsonResource.java [hunk 1]
```diff
@@ -195,4 +196,16 @@ public void clear() {
     public String toString() {
         return this.resource.toString();
     }
+    
+    @Override
+    public boolean equals(final Object other) {
```
Comment (llorllale) [inline src/main/java/com/amihaiemil/docker/JsonResource.java]: @amihaiemil do we need equals() and hashcode()?

File: src/main/java/com/amihaiemil/docker/Combined.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+
+/**
+ * Combine more JsonObjects into a single one.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.0.2
+ */
+final class Combined extends JsonResource {
```
Comment (llorllale) [inline src/main/java/com/amihaiemil/docker/Combined.java]: @amihaiemil I think `Merged` sounds better

Comment (amihaiemil) [reply src/main/java/com/amihaiemil/docker/Combined.java]: @llorllale Sure, I'll rename it

**Comment by amihaiemil** [comment]
@llorllale done

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31):

**Comment by 0crat** [comment]
The job [#139](https://github.com/amihaiemil/docker-java-api/issues/139) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**